doc/api/n-api.md

```diff
@@ -0,0 +1,3151 @@
+an exception was thrown, the `napi_status` value for the error
+will be returned.  If an exception was thrown, and no error occurred,
+`napi_pending_exception` will be returned.
+
+In cases where a return value other than `napi_ok` or
+`napi_pending_exception` is returned, you must call
+`napi_is_exception_pending()` to check if an exception is pending.
+See the section on exceptions for more details.
+
+The full set of possible napi_status values is defined
+in `napi_api_types.h`.
+
+Thr `napi_status` return value provides a vm-independent representation of
+the error which occurred.  In some cases it is useful to be able to get
+more detailed information, including a string representing the error as well as
+vm(engine)-specific information.  **NOTE:**  you should not rely on the
+content or format of any of the extended information as it is
+not subject to SemVer and may change at any time. It is intended
+only for logging purposes.
+
+In order to retrieve this information, the following method is provided:
+
+```C
+NODE_EXTERN const napi_extended_error_info* napi_get_last_error_info();
+```
+and the format of the `napi_extended_error_info` structure is as follows:
+
+```C
+struct napi_extended_error_info {
+  const char* error_message;
+  void* engine_reserved;
+  uint32_t engine_error_code;
+  napi_status error_code;
+};
+```
+
+`napi_get_last_error_info()` returns the information for the last
+N-API call that was made and the  `error_code` will match the
+`napi_status` returned by that call.  The `error_message` will
+be a textual representation of the error that occurred.
+
+
+#### napi_get_last_error_info
+##### Signature
+```C
+NAPI_EXTERN napi_status
+napi_get_last_error_info(napi_env env,
+                         const napi_extended_error_info** result);
+```
+##### Parameters
+- `[in]  env `: The environment that the API is invoked under
+- `[out]  result`: The `napi_extended_error_info` structure with more
+information about the error.
+
+##### Return value
+- `napi_ok` if the API succeeded.
+
+##### Description
+This API retrieves a `napi_extended_error_info` structure with information
+about the last error tha occured.
+
+
+### Exceptions
+
+Any N-API function may result in a pending JavaScript exception.  This is
+obviously the case for any function that may cause the execution of JavaScript
+but N-API specifies that an exception may be pending on return from any
+of the API functions.
+
+If the `napi_status` returned by a function is `napi_ok` then no
+exception is pending and no additional action is required.  If the
+`napi_status` returned is anything other than `napi_ok` or
+`napi_pending_exception`, and you want to try to recover and continue
+instead of simply returning immediately, then you must call the
+following function in order to determine if an exception is pending or not:
+
+```C
+napi_is_exception_pending()
+```
+
+and then check the value of `result`.
```

Comment:
This could probably be combined with the last sentence of the previous paragraph.